opentelemetry-sdk/src/logs/log_processor.rs

```diff
@@ -812,6 +812,26 @@ mod tests {
         assert_eq!(1, exporter.get_emitted_logs().unwrap().len())
     }
 
+
+    #[tokio::test(flavor = "current_thread")]
+    async fn test_batch_log_processor_shutdown_with_async_runtime() {
+        let exporter = InMemoryLogsExporterBuilder::default()
+            .keep_records_on_shutdown()
+            .build();
+        let processor = BatchLogProcessor::new(
+            Box::new(exporter.clone()),
+            BatchConfig::default(),
+            runtime::Tokio,
```

Comment:
With `Tokio` this probably will cause a deadlock. By design this should work with `TokioCurrentThread`

Reply:
@ThomsonTan Can we add a test for these combinations - with/without `current_thread` annotation, and using `runtime::TikioCurrentThread` ?

Reply:
Added 3 more tests to cover the combination of flavor and our runtime parameter.